lldb/source/Target/Thread.cpp

```diff
@@ -462,6 +462,10 @@ void Thread::ResetStopInfo() {
 }
 
 void Thread::SetStopInfo(const lldb::StopInfoSP &stop_info_sp) {
+  if (stop_info_sp &&
+      stop_info_sp->GetStopReason() == lldb::eStopReasonBreakpoint)
+    SetThreadHitBreakpointSite();
```

Comment:
Seems like you could move this after the `m_stop_info_sp` check a few lines below.